influxdb3_server/src/auth.rs

```diff
@@ -0,0 +1,60 @@
+use async_trait::async_trait;
+use authz::{Authorizer, Error, Permission};
+use observability_deps::tracing::{debug, warn};
+use sha2::{Digest, Sha512};
+
+/// An [`Authorizer`] implementation that will grant access to all
+/// requests that provide `token`
+#[derive(Debug)]
+pub struct AllOrNothingAuthorizer {
+    token: Vec<u8>,
+}
+
+impl AllOrNothingAuthorizer {
+    pub fn new(token: Vec<u8>) -> Self {
+        Self { token }
+    }
+}
+
+#[async_trait]
+impl Authorizer for AllOrNothingAuthorizer {
+    async fn permissions(
+        &self,
+        token: Option<Vec<u8>>,
+        perms: &[Permission],
+    ) -> Result<Vec<Permission>, Error> {
+        debug!(?perms, "requesting permissions");
+        let Some(provided) = token.as_deref() else {
+            return Err(Error::NoToken);
+        };
```

Comment:
I would just condense this down to a single line like so:

```suggestion
        let provided = token.as_deref().ok_or(Error::NoToken)?;
```